### PR TITLE
updated Contributing.md

### PR DESCRIPTION
**Pull request by Huzaif-Ahmed**
## Description
I've made comprehensive updates to the `contributing.md` file, detailing how to set up Firebase emulators and connect them to the cloud web app. These changes aim to simplify the onboarding process for contributors and ensure a smoother development experience.

## Related Issue
[Link to Issue #874](https://github.com/scorelab/Codelabz/issues/874)

## Motivation and Context
I referred to the official Firebase Emulator Suite documentation [here](https://firebase.google.com/docs/emulator-suite/install_and_configure) to ensure the accuracy and completeness of the setup instructions provided in `contributing.md`. These changes will help new and existing contributors quickly get started with Firebase emulators within the project.

## Checklist:
- [x] I have updated the documentation accordingly.

These updates are designed to enhance the project's documentation and contribute to a more welcoming and accessible environment for contributors.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -85,6 +85,36 @@ This will setup your project along with firebase emulator in a docker environmen
 
 You should fill in these values in their relevant fields in the `.env` file.
 
+
+### Firebase Emulator Setup
+
+From your parent directly run cd/functions
+1. run
```
Comment (shivareddy6) [inline CONTRIBUTING.md:92]: I believe this step is not required, can you re-check if this is mandatory. I dont remember running this while setting up in my local machine.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -85,6 +85,36 @@ This will setup your project along with firebase emulator in a docker environmen
...
+```
+shell npm init
+ ```
+2. refer this site [https://firebase.google.com/docs/emulator-suite/install_and_configure]
+3. make sure you have the correct jdk version installed
+4. then go to paren directory bu running command (```cd ..```)
+5. now lets connect your local firebase to cloub by running command (```firebase login```)
+6. then authenticate your firebase using browser and set the selected web app for codelabz
+7. then run the command (```firebase init```)
+8. select all the emulator necessitites by pressing a or selecting them amnually and pressing space
```
Comment (shivareddy6) [inline CONTRIBUTING.md:102]: You miss spelled **manually**, Can you change that.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -85,6 +85,36 @@ This will setup your project along with firebase emulator in a docker environmen
...
+9. answer the commands
+10. lets set up your credentials of test data
+11. then run command
```
Comment (shivareddy6) [inline CONTRIBUTING.md:105]: Can you mention the **make** command here and also mention if make is not present then run the following command.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -85,6 +85,36 @@ This will setup your project along with firebase emulator in a docker environmen
...
+```
+firebase emulators:start --import=testdata
+```
+12. if data is already present then run run
```
Comment (shivareddy6) [inline CONTRIBUTING.md:109]: Can you change this step to "If you want to start without any testdata, use the following command"

**Comment by Huzaif-Ahmed** [comment]
made the suggested changes can you please check it

**Pull request closed without merging**